lte/gateway/c/session_manager/ChargingGrant.cpp

```diff
@@ -80,12 +80,14 @@ CreditValidity ChargingGrant::is_valid_credit_response(
   bool gsu_all_invalid =
       !gsu.total().is_valid() && !gsu.rx().is_valid() && !gsu.tx().is_valid();
   if (gsu_all_invalid) {
-    if (update.credit().is_final()) {
+    if (update.credit().is_final() || credit_validity == TRANSIENT_ERROR) {
```

Comment:
in case of `TRANSIENT_ERROR` we will install the credit, no matter if GSU is empty

We may need to reconsidering removing `gsu_all_invalid` condition in near future